marlowe-playground-client/src/Simulation/State.purs

```diff
@@ -202,20 +182,61 @@ _payments = _log <<< to (mapMaybe f)
 
   f (OutputEvent _ payment) = Just payment
 
-emptyMarloweState :: Slot -> MarloweState
-emptyMarloweState sn =
+emptyExecutionStateWithSlot :: Slot -> ExecutionState
+emptyExecutionStateWithSlot sn =
   { possibleActions: mempty
   , pendingInputs: mempty
   , transactionError: Nothing
   , transactionWarnings: []
+  , log: []
   , state: emptyState sn
   , slot: sn
   , moneyInContract: mempty
-  , contract: Nothing
+  }
+
+type MarloweState
+  = { executionState :: Maybe ExecutionState
+    , contract :: Maybe Contract
+    , editorErrors :: Array IMarker
+    , editorWarnings :: Array IMarker
+    , holes :: Holes
+    }
+
+_executionState :: forall s a. Lens' { executionState :: a | s } a
+_executionState = prop (SProxy :: SProxy "executionState")
+
+_contract :: forall s a. Lens' { contract :: a | s } a
+_contract = prop (SProxy :: SProxy "contract")
+
+_editorErrors :: forall s a. Lens' { editorErrors :: a | s } a
+_editorErrors = prop (SProxy :: SProxy "editorErrors")
+
+_editorWarnings :: forall s a. Lens' { editorWarnings :: a | s } a
+_editorWarnings = prop (SProxy :: SProxy "editorWarnings")
+
+_holes :: forall s a. Lens' { holes :: a | s } a
+_holes = prop (SProxy :: SProxy "holes")
+
+--- Language.Haskell.Interpreter ---
+_result :: forall s a. Lens' { result :: a | s } a
+_result = prop (SProxy :: SProxy "result")
+
+emptyMarloweState :: MarloweState
+emptyMarloweState =
```

Comment:
I think `emptyMarloweState = mempty`

Reply:
`  No type class instance was found for Data.Semigroup.Semigroup Contract`